### ISSUE TITLE
BGPeer incorrectly requiring values

### ISSUE THREAD

**Issue by univ0298** [projectcalico/libcalico-go#1390]
## Expected Behavior
```
calicoctl apply -f - <<EOF
kind: BGPPeer
apiVersion: projectcalico.org/v3
metadata:
  name: peer-with-route-reflectors
spec:
  nodeSelector: all()
  peerSelector: route-reflector == 'true'
EOF
Failed to apply 'BGPPeer' resource: [BGPPeer.crd.projectcalico.org "peer-with-route-reflectors" is invalid: [spec.asNumber: Required value, spec.peerIP: Required value]]
```

This should work and is even documented here: https://docs.projectcalico.org/networking/bgp . It looks like some change has broken the required values @caseydavenport 

## Current Behavior
It shows that asNumber and peerIP are required when it fact it's the opposite - they must NOT be provided when specifying a peerSelector

## Possible Solution
Perhaps related to this change? https://github.com/projectcalico/libcalico-go/commit/ff61058d907e4cfee18d45d6099dc0f07a1a1238

## Steps to Reproduce (for bugs)
Shown above

## Context
Unable to setup Calico!

## Your Environment
```
calicoctl version
Client Version:    3.18.1
Git commit:        911f383
Cluster Version:   v3.16.8
Cluster Type:      k8s,bgp,kdd,typha
```
MacOS
```
uname -a
Darwin richards-mbp.lan 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64
```

**Comment by univ0298** [comment]
Also probably related: https://github.com/projectcalico/libcalico-go/commit/8a0cf3af106ac9aef7c02e0230fc8dcc3c7e6597

**Comment by univ0298** [comment]
```
calicoctl version
Client Version:    3.18.1
Git commit:        911f383
Cluster Version:   v3.16.8
Cluster Type:      typha,kdd,k8s,bgp
```

Perhaps this is fixed if my cluster was at a later version?

**Comment by univ0298** [comment]
Looking at this more carefully it seems very likely it has been fixed in 3.18 by this commit:
https://github.com/projectcalico/libcalico-go/commit/ff61058d907e4cfee18d45d6099dc0f07a1a1238
